proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,177 @@
+---
+simd: '0296'
+title: Increase Transaction Size
+authors:
+  - jacobcreech
+  - apfitzge
+category: Standard
+type: Core
+status: Review
+created: 2025-05-28
+feature: (fill in with feature key and github tracking issues once accepted)
+---
+
+## Summary
+
+Historically, messages transmitted over the network must not exceed the IPc6 MTU
+size to ensure a fast and reliable network delivery. Solana has used a 
+conservative MTU size of 1280 bytes, which after accounting for some of the
+overhead, leaves a maximum transaction size of 1032 bytes for data in the
+payload.
+
+With QUIC, we can now use a larger transaction size than the original 1280
+bytes. This proposal presents a new transaction size starting at 4096 bytes,
+and the new v1 transaction format required to support the larger size. v1 
+transactions are a direct upgrade from the legacy format, while proposing 
+a removal of Address Lookup Tables compared to v0 transactions.
```

Comment:
Can we also drop durable nonces in v1?
The reason to remove ALT is to be able to validate a tx without hitting the account db. Durable nonces have the same issue, we can't validate a tx without hitting the account db. Hence the proposal to drop them.

Reply:
I'm open to it though if there's a path forward

Reply:
v0 and v1 will coexist in the near future, but ideally we'd deprecate and remove v0. It'll be a long migration though and we'd need a solution to nonce transactions before we deprecate v0 and legacy.

Reply:
If we remove nonce support from the tx format, what else is needed to be done on v1 transactions? It seems like consensus from both Anza and FD is to remove in v1 so we have a migration path to remove nonces entirely in the future.